### ISSUE TITLE
Error Gnome 41

### ISSUE THREAD

**Issue by miles969** [dpoetzsch/keyman#15]
hi!
keyman fails to load on gnome 41 with a TypeError:

```
JS ERROR: Extension keyman@dpoetzsch.github.com: TypeError: 
can't redefine non-configurable property "define"
.../.local/share/gnome-shell/extensions/keyman@dpoetzsch.github.com/keyringDbus.js:8:8
.../.local/share/gnome-shell/extensions/keyman@dpoetzsch.github.com/data.js:9:17
.../.local/share/gnome-shell/extensions/keyman@dpoetzsch.github.com/keyman.js:22:14
.../.local/share/gnome-shell/extensions/keyman@dpoetzsch.github.com/extension.js:12:16
_callExtensionInit@resource:///org/gnome/shell/ui/extensionSystem.js:421:13
loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:346:27
_onInstallButtonPressed@resource:///org/gnome/shell/ui/extensionDownloader.js:272:35
Async*addButton/<@resource:///org/gnome/shell/ui/dialog.js:136:41
```

**Comment by dpoetzsch** [comment]
Thanks for your report!

This seems to be a problem with the commonjs implementation of https://github.com/darkoverlordofdata/gir2dts and gnome 41. Maybe Gnome 41 finally has a module loader integrated into core?

I don't have gnome 41 myself yet but this does not look like I can just do a quick fix for this. I'll have to find time to look at it.

**Comment by rpkamp** [comment]
I just switched to Gnome 41 and didn't get that error. I did get a very similar error, but about "console".

So I removed the bit about console from `keyringDbus.js` (https://github.com/dpoetzsch/keyman/blob/master/keyringDbus.js#L35-L40), bumped the Gnome version to 41 in `metadata.json` and the extension is working as it should now for me (using `./deploy.sh`, logging out and logging back in).

I'm running Fedora 35 with GNOME Shell 41.2 (gjs 1.70.0)

**Comment by negesti** [comment]
Just upgrade to Fedora 35 too and had to comment the console lines in keyringDbus.js. 

` 34         })) },
 35     /**console: { value: {
 36             log: function () { print.apply(null, arguments); },
 37             warn: function () { print.apply(null, arguments); },
 38             error: function () { print.apply(null, arguments); },
 39             info: function () { print.apply(null, arguments); }
 40         } },**/
`

**Comment by omron93** [comment]
@rpkamp  is right and simple patch like this and `bash ./deploy.sh` works quite fine with Gnome 41.
```
diff --git a/keyringDbus.js b/keyringDbus.js
index 686c67d..eff58c0 100644
--- a/keyringDbus.js
+++ b/keyringDbus.js
@@ -32,12 +32,12 @@ Object.defineProperties(window, {
             Pango: { id: 'Pango', exports: imports.gi.Pango },
             GObject: { id: 'GObject', exports: imports.gi.GObject }
         })) },
-    console: { value: {
-            log: function () { print.apply(null, arguments); },
-            warn: function () { print.apply(null, arguments); },
-            error: function () { print.apply(null, arguments); },
-            info: function () { print.apply(null, arguments); }
-        } },
+//    console: { value: {
+//            log: function () { print.apply(null, arguments); },
+//            warn: function () { print.apply(null, arguments); },
+//            error: function () { print.apply(null, arguments); },
+//            info: function () { print.apply(null, arguments); }
+//        } },
     _: { value: imports.gettext.gettext }
 });
 Object.defineProperties(define, {
diff --git a/metadata.json b/metadata.json
index 5546697..323e563 100644
--- a/metadata.json
+++ b/metadata.json
@@ -1,5 +1,5 @@
 {
-  "shell-version": ["40"],
+  "shell-version": ["41"],
   "original-authors": ["keyman@poehe.de"],
   "uuid": "keyman@dpoetzsch.github.com",
   "name": "KeyMan",
```
@dpoetzsch Can I create a PR to have this working out of the box? or how to proceed to get this issue fixed?

**Comment by dpoetzsch** [comment]
TLDR: I'd be happy to accept a PR and release a new version.

---

Sorry for the delay, I've been quite busy with work recently.

I think the proper way to fix this would be to fix it in https://github.com/darkoverlordofdata/gir2dts/blob/master/src/gjs.js, release a new version of gir2dts (which is out of my hands). Another, maybe even better way would be to replace gir2dts with the newer and seemingly more active [ts-for-gir](https://github.com/sammydre/ts-for-gir). Then, use the new fixed type bindings to create a new version of https://github.com/dpoetzsch/gnome-keyring-js/.

That being said, I want this plugin to just work for people, so I'd be happy to accept a PR in the meantime and release that if it fixes the problem. I can go the long way when I find time for it.

**Comment by dpoetzsch** [comment]
@omron93 I just upgraded to Fedora 35 and the patch you suggested works wonderfully. I already applied it in the repo and will publish a new version now. So no need for a PR :-)

**Comment by omron93** [comment]
Perfect. Thanks for quick fix
